core/src/test/java/org/hibernate/ogm/utils/OgmTestRunner.java

```diff
@@ -140,6 +140,8 @@ protected void runChild(FrameworkMethod method, RunNotifier notifier) {
 		}
 		finally {
 			if ( testMethodScopedSessionFactory != null ) {
+				cleanUpPendingTransactionIfRequired();
+				TestHelper.dropSchemaAndDatabase( testScopedSessionFactory );
 				testMethodScopedSessionFactory.close();
```

Comment:
Nice one.

It seems to me there are a lot of commits in this PR that are unrelated to the grouping and that could be considered as separate issues. Since I tend to do the same thing, it makes me wonder if there is something that we could do to improve the creation of separate issues (make it faster). This way PR requests would be shorter and less intimidating.

By thew way, I'm just thinking out loud, it's on a critic on your PR